src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlObjectSerializerReadContextComplexJson.cs

```diff
@@ -48,22 +48,28 @@ public int GetJsonMemberIndex(XmlReaderDelegator xmlReader, XmlDictionaryString[
             int length = memberNames.Length;
             if (length != 0)
             {
-                for (int i = 0, index = (memberIndex + 1) % length; i < length; i++, index = (index + 1) % length)
+                for (int i = 0, index = (memberIndex + 1) % length; i < length; i++)
```

Comment:
There's also the pattern in the first part of the for loop.
I believe System.Private.DataContractSerialization is in low maintenance mode that only security fix will be considered. Can any area owner confirm that?

Reply:
c.c. @StephenMolloy